src/qtcore/qml/elements/QtQuick/Row.js

```diff
@@ -9,9 +9,9 @@ registerQmlType({
 function QMLRow(meta) {
     QMLPositioner.call(this, meta);
 
-    createProperty({ type: "enum", object: this, name: "layoutDirection" });
+    createProperty({ type: "enum", object: this, name: "layoutDirection", initialValue: 0 });
     this.layoutDirectionChanged.connect(this, this.layoutChildren);
-    this.layoutDirection = 0;
+    this.layoutDirectionChanged(0);
```

Comment:
I'm trying to preserve the original behavior as best as I can.
At some places, the default value for the property were only set after connecting functions to the property's changed signal. In those cases, I've made sure the signals were still going to be triggered.
I'm not sure it does something in every cases : but I know that at least in some, it is important (for instance, I forgot it at some point on the Image item, and it broke an image test).